core/src/test/scala/integration/kafka/server/DynamicBrokerReconfigurationTest.scala

```diff
@@ -452,20 +556,35 @@ class DynamicBrokerReconfigurationTest extends ZooKeeperTestHarness with SaslSet
 
   private class ConsumerThread(producerThread: ProducerThread) extends ShutdownableThread("test-consumer", isInterruptible = false) {
     private val consumer = createConsumer("group1", trustStoreFile1)
+    @volatile var lastBatch: ConsumerRecords[String, String] = _
     @volatile private var endTimeMs = Long.MaxValue
     var received = 0
     override def doWork(): Unit = {
       try {
         while (isRunning || (received < producerThread.sent && System.currentTimeMillis < endTimeMs)) {
-          received += consumer.poll(50).count
+          val records = consumer.poll(50)
+          received += records.count
+          if (!records.isEmpty)
+            lastBatch = records
         }
       } finally {
         consumer.close()
       }
     }
+
     override def initiateShutdown(): Boolean = {
       endTimeMs = System.currentTimeMillis + 10 * 1000
       super.initiateShutdown()
     }
+
+    def waitForRecords(f: ConsumerRecord[String, String] => Boolean): Unit = {
```

Comment:
nit: maybe `predicate` is a better name for the argument? Also, maybe rename to `waitForRecordsMatching`?